src/mono/browser/runtime/startup.ts

```diff
@@ -216,7 +221,10 @@ async function onRuntimeInitializedAsync(userOnRuntimeInitialized: () => void) {
         await runtimeHelpers.afterPreRun.promise;
         mono_log_debug("onRuntimeInitialized");
 
-        runtimeHelpers.mono_wasm_exit = cwraps.mono_wasm_exit;
+        runtimeHelpers.mono_wasm_exit = !MonoWasmThreads ? cwraps.mono_wasm_exit : (code) => {
```

Comment:
General thought: I believe we should avoid adding behavior just to functions added to helpers.